testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/ConflictingScopePermissionTest.java

```diff
@@ -111,6 +164,59 @@ public void testMartaCanAccessResourceAWithExecuteAndWrite() {
                     assertTrue(permission.getScopes().contains("read"));
                     permissions.remove(permission);
                     break;
+                case "Resource B":
+                    assertEquals(3, permission.getScopes().size());
+                    assertTrue(permission.getScopes().contains("execute"));
+                    assertTrue(permission.getScopes().contains("write"));
+                    assertTrue(permission.getScopes().contains("read"));
+                    permissions.remove(permission);
+                    break;
+                default:
+                    fail("Unexpected permission for resource [" + resourceSetName + "]");
+            }
+        }
+
+        assertTrue(permissions.isEmpty());
+    }
+
+    @Test
+    public void testWithDisabledMode() throws Exception {
+        ClientResource client = getClient(getRealm());
+        AuthorizationResource authorization = client.authorization();
+        ResourceServerRepresentation settings = authorization.getSettings();
+
+        settings.setPolicyEnforcementMode(PolicyEnforcementMode.DISABLED);
+
+        authorization.update(settings);
+
+        Collection<Permission> permissions = getEntitlements("marta", "password");
+
+        assertEquals(3, permissions.size());
+
+        for (Permission permission : new ArrayList<>(permissions)) {
+            String resourceSetName = permission.getResourceName();
+
+            switch (resourceSetName) {
+                case "Resource A":
+                    assertEquals(3, permission.getScopes().size());
+                    assertTrue(permission.getScopes().contains("execute"));
+                    assertTrue(permission.getScopes().contains("write"));
+                    permissions.remove(permission);
+                    break;
+                case "Resource C":
+                    assertEquals(3, permission.getScopes().size());
+                    assertTrue(permission.getScopes().contains("execute"));
+                    assertTrue(permission.getScopes().contains("write"));
+                    assertTrue(permission.getScopes().contains("read"));
+                    permissions.remove(permission);
+                    break;
+                case "Resource B":
+                    assertEquals(3, permission.getScopes().size());
```

Comment:
This and three following lines can be replaced by `assertThat(permission.getScopes(), containsInAnyOrder("execute", "write", "read"));` - improves message on failure.